sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -142,7 +142,7 @@ case class CreateTableAsSelect[T](
     tableName: String,
     child: LogicalPlan,
     allowExisting: Boolean,
-    desc: Option[T] = None) extends UnaryNode {
+    desc: T) extends UnaryNode {
```

Comment:
Let's get rid of the type parameter and rename it to `CreateHiveTableAsSelect` (be a little bit more specific on what this one does).